### ISSUE TITLE
UiPopover doesn't work if placed within a slot

### ISSUE THREAD

**Issue by CarlEdmondson** [JosephusPaye/Keen-UI#356]
The ui-popover component doesn't work when the component and it's trigger are placed within the slot of another component. I believe this is due to the fact that the trigger must be referenced in the parent of the ui-popover component.

However if you define the trigger element and the ui-popover in the slot for a component then the trigger is not actually in the $refs object of the ui-popover component's $parent.

I'm not sure what the solution to this could be without changing how the trigger is defined on the popover component.

**Comment by JosephusPaye** [comment]
Hi @CarlEdmondson,

You are right, we had to rewrite the way the trigger prop is handled.

From the docs:

> **trigger**: An Element or VueComponent instance to use for the trigger, or a CSS selector string.
> [...]
> If trigger is not provided or the selector doesn't match any element, the ripple's immediate parent element is used as the trigger. Note that the parent used is the ripple's parent in the DOM, not the Vue parent component.

This problem should now be fixed in `next`.

To try it:

```sh
# npm
npm install keen-ui@JosephusPaye/Keen-UI#next

# yarn
yarn add keen-ui@JosephusPaye/Keen-UI#next
```

See the [changelog](https://github.com/JosephusPaye/Keen-UI/blob/next/CHANGELOG.md) for all changes.

**Comment by JosephusPaye** [comment]
Released in [v1.1.0](https://github.com/JosephusPaye/Keen-UI/releases/tag/v1.1.0).
